### PR TITLE
List workflow optimization

### PR DESCRIPTION
**Pull request by SarthakJain26**
## Proposed changes

]Updated ListWorkflow API to fetch only requested data from the DB.
Added Pagination, Filtering, Sorting, and Searching at DB level for ListWorkflow API.
Updated frontend according to the API's.
Removed deprecated API's from backend and frontend.
Minor style changes due to litmus-ui version 1.1.7

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

## Special notes for your reviewer:
Please ignore translation if not important

### CODE REVIEW

**Review by arkajyotiMukherjee — APPROVED** [review]

**Review by Saranya-jena — APPROVED** [review]

**Review by ishangupta-ds — APPROVED** [review]

**Review by whitetiger1399 — APPROVED** [review]

File: litmus-portal/graphql-server/pkg/chaos-workflow/handler/handler.go [hunk 1]
```diff
@@ -400,78 +399,151 @@ func QueryWorkflowRuns(input model.GetWorkflowRunsInput) (*model.GetWorkflowsOut
...
+
+	var workflows []dbSchemaWorkflow.AggregatedWorkflows
+
+	if err = workflowsCursor.All(context.Background(), &workflows); err != nil {
+		return &model.ListWorkflowsOutput{
+			TotalNoOfWorkflows: 0,
+			Workflows:          result,
+		}, nil
+	}
+
```
Comment (rajdas98) [inline litmus-portal/graphql-server/pkg/chaos-workflow/handler/handler.go:534]: Please add a check condition for `workflows` here to catch out of bound issue

File: litmus-portal/graphql-server/pkg/chaos-workflow/handler/handler.go [hunk 2]
```diff
@@ -490,7 +562,17 @@ func QueryListWorkflow(project_id string, workflowIds []*string) ([]*model.Workf
 		}
 		result = append(result, &newChaosWorkflows)
 	}
-	return result, nil
+
+	totalFilteredWorkflows := 0
```
Comment (rajdas98) [inline litmus-portal/graphql-server/pkg/chaos-workflow/handler/handler.go:566]: Please rename it as totalFilteredWorkflowsCounter.

File: litmus-portal/graphql-server/pkg/chaos-workflow/handler/handler.go (hunk 1 above)
Comment (SarthakJain26) [reply litmus-portal/graphql-server/pkg/chaos-workflow/handler/handler.go:534]: Done

File: litmus-portal/graphql-server/pkg/chaos-workflow/handler/handler.go (hunk 2 above)
Comment (SarthakJain26) [reply litmus-portal/graphql-server/pkg/chaos-workflow/handler/handler.go]: Done

**Review by rajdas98 — APPROVED** [review]

**Pull request merged**